### PR TITLE
Add support for ramfs as well as tmpfs in volume mounts

### PR DESCRIPTION
**Pull request by rhatdan**
Users want to mount a tmpfs file system with secrets, and make
sure the secret is never saved into swap. They can do this either
by using a ramfs tmpfs mount or by passing `noswap` option to
a tmpfs mount.

Fixes: https://github.com/containers/podman/issues/19659

#### Does this PR introduce a user-facing change?

```release-note
podman run and create --mount now support the `ramfs` type.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Review by flouthoc — APPROVED** [review]
LGTM probably buildah will need this as well ?

**Comment by flouthoc** [comment]
@giuseppe PTAL

File: pkg/specgenutil/volumes.go
Comment (vrothberg) [inline pkg/specgenutil/volumes.go:529]: Can we rename the function to `parseMount` or something like that?  It's not strictly `Tmpfs` only anymore.

**Review by vrothberg — APPROVED** [review]
Can you add a commit message as well?

> Add support for ramfs as well as tmpfs in volume mounts 

I am confused by the title since `tmpfs` mounts were supported prior to this change

File: pkg/specgenutil/volumes.go
Comment (rhatdan) [reply pkg/specgenutil/volumes.go]: Add noswap, but this seems like it is only supported in rootful mode.

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

Thanks!

**Comment by mheon** [comment]
LGTM
/hold cancel

**Pull request merged**
